traffic_ops/traffic_ops_golang/cdnfederation/cdnfederations.go

```diff
@@ -143,7 +144,18 @@ func checkTenancy(tenantID *int, tenantIDs []int) bool {
 	return false
 }
 
-func (fed *TOCDNFederation) Read() ([]interface{}, error, error, int) {
+func (fed *TOCDNFederation) Read(h map[string][]string) ([]interface{}, error, error, int) {
+	ims := h["If-Modified-Since"]
+	var modifiedSince time.Time
+	modified := false
+	found := false
+	code := http.StatusOK
+
+	if ims != nil && len(ims) != 0 {
+		if t, err := time.Parse(time.RFC1123, ims[0]); err == nil {
```

Comment:
It seems that nearly every endpoint needs this. Should we reduce the duplicated logic by having the HTTP Routing parse this and put it in the HTTP Context?